### DISCUSSION TITLE
`Most Used Lang` Showing Wrong Output

### DISCUSSION THREAD

**Discussion by BIGBEASTISHANK** [anuraghazra/github-readme-stats#3705] [Q&A] [answered]
### Describe the bug

First, I am pretty sure that I use more JavaScript than CSS, and my reading shows that I use more CSS, HTML, and C# than JS. Even if it might occur due to my dotfile, how is it less than HTML? I mostly have JS repos, and my website has 6 versions, and I don't even have a proper repo with HTML. I don't even get the option to sort by html on my repo tab. Second, it doesn't even show my C++ language. I have 2 repositories of C++ languages: 1 is archived and one is running. It is not even showing C++, not matter what.

The excluded languages works fine
`&hide=shell,hlsl,shaderlab,glsl,qml,lua`

### Screenshots / Live demo link

My Card
![image](https://github.com/anuraghazra/github-readme-stats/assets/77890275/c26e912c-5969-4337-bd04-79eed7b12499)

Language sorter
![image](https://github.com/anuraghazra/github-readme-stats/assets/77890275/49ab5502-e3ab-4867-8a02-d278049c02a8)

**Comment by qwerty541** [comment]
# Languages calculation algorithm

By default we calculate languages percentages by files size in bytes. There is also ability to calculate languages percentages by count of repositories where the particular language is used. Also you can combine there two options. You can customize how languages algorithm works with `size_weight` and `count_weight` query string parameters. Here is detailed explanation of how it works in our documentation https://github.com/anuraghazra/github-readme-stats?tab=readme-ov-file#language-stats-algorithm. May be calculation by repositories count will be more suiteable for you.

# Deployment

I see that currently you are using `https://github-readme-stats-anuraghazra1.vercel.app` deployment in you GitHub profile readme. Be aware that it is NOT our official deployment and noone know whether it's owner keep it up to date or not, not even speaking about that there can be deployed someones custom branch. so there is no sense to report problems with it. Currently this deployment shows different result in comparison with our public deployment `https://github-readme-stats.vercel.app/`. In particular, public deployment shows usage of C++ and Python. To summarize, I recommend you to use public deployment which always uses latest code or [deploy own Vercel instance](https://github.com/anuraghazra/github-readme-stats?tab=readme-ov-file#on-vercel).

# Examples

Official deployment, percentages by bytes count.

```html
<img src="https://github-readme-stats.vercel.app/api/top-langs/?username=bigbeastishank&theme=dark&hide_border=false&hide=shell,hlsl,shaderlab,glsl,qml,lua&langs_count=20" />
```

<img src="https://github-readme-stats.vercel.app/api/top-langs/?username=bigbeastishank&theme=dark&hide_border=false&hide=shell,hlsl,shaderlab,glsl,qml,lua&langs_count=20" />

Official deployment, percentages by repositories count.

```html
<img src="https://github-readme-stats.vercel.app/api/top-langs/?username=bigbeastishank&theme=dark&hide_border=false&hide=shell,hlsl,shaderlab,glsl,qml,lua&langs_count=20&size_weight=0&count_weight=1" />
```

<img src="https://github-readme-stats.vercel.app/api/top-langs/?username=bigbeastishank&theme=dark&hide_border=false&hide=shell,hlsl,shaderlab,glsl,qml,lua&langs_count=20&size_weight=0&count_weight=1" />

**Reply by BIGBEASTISHANK** [reply]
Thanks for the help

**Reply by BIGBEASTISHANK** [reply]
Hey, after doing `https://github-readme-stats.vercel.app/api/top-langs/?username=bigbeastishank&theme=dark&hide_border=false&no-bg=true&no-frame=true&hide=shell,hlsl,shaderlab,glsl,qml,lua&langs_count=20&size_weight=0&count_weight=1&exclude_repo=dotfiles` I am counting it by repos, and then my CSS gets highest after I exclude the only repo that has CSS marked. (ie. dotfiles)

```html
<img src="https://github-readme-stats.vercel.app/api/top-langs/?username=bigbeastishank&theme=dark&hide_border=false&no-bg=true&no-frame=true&hide=shell,hlsl,shaderlab,glsl,qml,lua&langs_count=20&size_weight=0&count_weight=1&exclude_repo=dotfiles" />
```
<img src="https://github-readme-stats.vercel.app/api/top-langs/?username=bigbeastishank&theme=dark&hide_border=false&no-bg=true&no-frame=true&hide=shell,hlsl,shaderlab,glsl,qml,lua&langs_count=20&size_weight=0&count_weight=1&exclude_repo=dotfiles" />


And if using the default language, it is showing HTML, which is very, very small, like 1 page with a few 50 lines of code. While JavaScript has very large file size, infact fact, JS is the biggest file on my github profile (if we exclude dotfiles)

```html
<img src="https://github-readme-stats.vercel.app/api/top-langs/?username=bigbeastishank&theme=dark&hide_border=false&hide=shell,hlsl,shaderlab,glsl,qml,lua&langs_count=20&exclude_repo=dotfiles" />
```
<img src="https://github-readme-stats.vercel.app/api/top-langs/?username=bigbeastishank&theme=dark&hide_border=false&hide=shell,hlsl,shaderlab,glsl,qml,lua&langs_count=20&exclude_repo=dotfiles" />

**Reply by qwerty541** [reply]
> Hey, after doing https://github-readme-stats.vercel.app/api/top-langs/?username=bigbeastishank&theme=dark&hide_border=false&no-bg=true&no-frame=true&hide=shell,hlsl,shaderlab,glsl,qml,lua&langs_count=20&size_weight=0&count_weight=1&exclude_repo=dotfiles I am counting it by repos, and then my CSS gets highest after I exclude the only repo that has CSS marked. (ie. dotfiles)

GitHub mark repositories by the most used language, but it does not mean that this language will be the only one used in the repository. When counting percentages by repositories, we check how much of your repositories contains CSS code, even if it just 1 byte. It turns that CSS is the most common language in your repositories.

> And if using the default language, it is showing HTML, which is very, very small, like 1 page with a few 50 lines of code. While JavaScript has very large file size, infact fact, JS is the biggest file on my github profile (if we exclude dotfiles)

You're wrong, there is three of your repositories with a lot of HTML code.

- https://github.com/BIGBEASTISHANK/Drag-Your-Tires
- https://github.com/BIGBEASTISHANK/Project-E23
- https://github.com/BIGBEASTISHANK/Initial-Websites

And there is also just 3 of 19 your repositories contain Javascript code, with much less files size.

I'm absolutely sure that there is no error, stats card showing all correct.

**Reply by BIGBEASTISHANK** [reply]
Ok so why does excluding dotfile decreases my cpp file from python?, like removing dot file should calculate less css, scss, qml, python

with dotfile
```html
<img src="https://github-readme-stats.vercel.app/api/top-langs/?username=bigbeastishank&theme=dark&hide_border=false&no-bg=true&no-frame=true&hide=shell,hlsl,shaderlab,glsl,qml,lua&langs_count=20&size_weight=0&count_weight=1&exclude_repo=" />
```
<img src="https://github-readme-stats.vercel.app/api/top-langs/?username=bigbeastishank&theme=dark&hide_border=false&no-bg=true&no-frame=true&hide=shell,hlsl,shaderlab,glsl,qml,lua&langs_count=20&size_weight=0&count_weight=1&exclude_repo=" />

without dotfile
```html
<img src="https://github-readme-stats.vercel.app/api/top-langs/?username=bigbeastishank&theme=dark&hide_border=false&no-bg=true&no-frame=true&hide=shell,hlsl,shaderlab,glsl,qml,lua&langs_count=20&size_weight=0&count_weight=1&exclude_repo=dotfiles" />
```
<img src="https://github-readme-stats.vercel.app/api/top-langs/?username=bigbeastishank&theme=dark&hide_border=false&no-bg=true&no-frame=true&hide=shell,hlsl,shaderlab,glsl,qml,lua&langs_count=20&size_weight=0&count_weight=1&exclude_repo=dotfiles" />

Like removing dotfiles increases my css and java script? but my dotfile only contain css, scss, qml, python, so how removing it increases them insted of decreasing it?

**Reply by qwerty541** [reply, marked as answer]
> Like removing dotfiles increases my css and java script? but my dotfile only contain css, scss, qml, python, so how removing it increases them insted of decreasing it?

Looks like you misunderstanding how percentages works. I will try to explain on example.

Image that you have 10 repositories and 1 of them contains Javascript code. At this moment you have 1 / 10 * 100 = 10% of your total repositories containing Javascript.

On next step we delete two random repositories, but not the one which contain Javascript. At this moment you have 1 / 8 * 100 = 12,5% of your total repositories containing Javascript.

Count of repositories with Javascript code still same but it takes more percentages since total count of your repositories decreased.

The same situation with your actual GitHub repositories and increased CSS and Javascript percentages.

The second reason of higher percentages is that percentages distributed between different count of languages on your two last examples.

**Reply by BIGBEASTISHANK** [reply]
Thanks I understood
